_src/_plugins/tufte.rb

```diff
@@ -0,0 +1,50 @@
+module Jekyll
+  class MarginNotes < Liquid::Block
+    alias_method :render_block, :render
+    @@margincount = 0;
+
+    def initialize(tag_name, reference, tokens)
+      super
+      @reference = reference
+      @@margincount += 1;
+      if @reference !~ /[^[:space:]]/
+        @reference = "marginnote-#{@@margincount}";
+      end
+    end
+
+    def render(context)
+      site = context.registers[:site]
+      converter = site.find_converter_instance(::Jekyll::Converters::Markdown)
+      content = converter.convert(super(context))
+      content.gsub! '<p>', ''
+      content.gsub! '</p>', ''
+      "<label for=\"#{@reference}\" class=\"margin-toggle\">&#8853;</label><input type=\"checkbox\" id=\"#{@reference}\" class=\"margin-toggle\"/><span class=\"marginnote\">#{content}</span>"
+    end
+  end
+
+  class SideNotes < Liquid::Block
+    alias_method :render_block, :render
+    @@sidenotecount = 0;
+
+    def initialize(tag_name, reference, tokens)
+      super
+      @reference = reference
+      @@sidenotecount += 1;
```

Comment:
Class variables are frowned on in Ruby. Won't this cause problems if the posts are built in a different order?

Imagine a situation like the following:

* post 1
  * sidenote 1
  * sidenote 2

* post 2
  * sidenote 1
  * sidenote 2

Won't post 2 have its sidenotes numbered "3" and "4"? If so, now what happens if post 2 is built first next time? Won't that renumber the notes and break any internal references?